src/RenderWebGL.js

```diff
@@ -1152,63 +1141,70 @@ class RenderWebGL extends EventEmitter {
             return [];
         }
 
-        // Only draw to the size of the untransformed drawable.
-        const gl = this._gl;
-        twgl.bindFramebufferInfo(gl, this._queryBufferInfo);
-        gl.viewport(0, 0, width, height);
-
-        // Clear the canvas with RenderConstants.ID_NONE.
-        const noneColor = Drawable.color4fFromID(RenderConstants.ID_NONE);
-        gl.clearColor.apply(gl, noneColor);
-        gl.clear(gl.COLOR_BUFFER_BIT);
-
-        // Overwrite the model matrix to be unrotated, unscaled, untranslated.
-        const modelMatrix = twgl.m4.identity();
-        twgl.m4.rotateZ(modelMatrix, Math.PI, modelMatrix);
-        twgl.m4.scale(modelMatrix, [width, height], modelMatrix);
-
-        const projection = twgl.m4.ortho(-0.5 * width, 0.5 * width, -0.5 * height, 0.5 * height, -1, 1);
-
-        this._drawThese([drawableID],
-            ShaderManager.DRAW_MODE.silhouette,
-            projection,
-            {extraUniforms: {u_modelMatrix: modelMatrix}}
-        );
-
-        const pixels = new Uint8Array(Math.floor(width * height * 4));
-        gl.readPixels(0, 0, width, height, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
-
-        // Known boundary points on left/right edges of pixels.
-        const boundaryPoints = [];
-
-        /**
-         * Helper method to look up a pixel.
-         * @param {int} x X coordinate of the pixel in `pixels`.
-         * @param {int} y Y coordinate of the pixel in `pixels`.
-         * @return {int} Known ID at that pixel, or RenderConstants.ID_NONE.
-         */
-        const _getPixel = (x, y) => {
-            const pixelBase = Math.round(((width * y) + x) * 4); // Sometimes SVGs don't have int width and height
-            return Drawable.color3bToID(
-                pixels[pixelBase],
-                pixels[pixelBase + 1],
-                pixels[pixelBase + 2]);
+        const CCW = function (A, B, C) {
+            return (((B[0] - A[0]) * (C[1] - A[1])) - ((B[1] - A[1]) * (C[0] - A[0])));
         };
-        for (let y = 0; y <= height; y++) {
+
+        // https://github.com/LLK/scratch-flash/blob/dcbeeb59d44c3be911545dfe54d
+        // 46a32404f8e69/src/scratch/ScratchCostume.as#L369-L413 Following
+        // RasterHull creation, compare and store left and right values that
+        // maintain a convex shape until that data can be passed to `hull` for
+        // further work.
```

Comment:
I'm not sure anyone on the team understands this algorithm any more; it looks like it was copied into the older Scratch codebase from Numerical Recipes or something. Anyway, I guess this is another "let's revisit this post-release": I think we should try to make this code readable to the average Scratch open-source contributor.

Reply:
I commented the code below as a first step to make it more understandable.

Reply:
Awesome - thanks!